### PR TITLE
Fix: Convert Filter Products by Price Block to TypeScript

### LINKED ISSUE #6495: Convert Filter Products by Price Block to TypeScript

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #6495

This PR converts the codebase of the Filter Products by Price block to TypeScript. This also fixes the props' types of the `<PriceSlider>` component.

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) .
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md)
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Screenshots

N/A

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

Ensure no regression for Filter Products by Price block after the conversion.

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: assets/js/blocks/price-filter/block.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			"default": 3
+		},
+		"className": {
+			"type": "string",
+			"default": ""
+		}
+	},
+	"textdomain": "woo-gutenberg-products-block",
+	"apiVersion": 2,
+    "$schema": "https://schemas.wp.org/trunk/block.json"
```
Comment (albarin) [inline assets/js/blocks/price-filter/block.json]: Could we also add this `$schema` property to the other `block.json` files in the PR?
Also, super picky, but I see a different indentation at the beginning of this line 🤔

**Review by albarin — CHANGES_REQUESTED** [review]
Looking good, left just a small comment.

File: assets/js/blocks/price-filter/block.json (hunk 1 above)
Comment (dinhtungdu) [reply assets/js/blocks/price-filter/block.json]: > Also, super picky, but I see a different indentation at the beginning of this line

Totally not! Thanks for pointing this out.

**Comment by dinhtungdu** [comment]
@albarin Thanks for the review! I updated the PR per your suggestion, please take another look at this!

**Review by albarin — APPROVED** [review]
👏

**Comment by dinhtungdu** [comment]
@albarin I updated some parts of the codebase to sync with the latest update and fix some additional compiler errors. Can you please take another look?

File: assets/js/blocks/price-filter/block.tsx [hunk 2]
```diff
@@ -116,12 +135,18 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
...
+			objectHasProp( results.price_range, 'min_price' ) &&
+			isString( results.price_range.min_price )
+				? results.price_range.min_price
+				: undefined,
+		maxPrice:
+			objectHasProp( results, 'price_range' ) &&
+			objectHasProp( results.price_range, 'max_price' ) &&
+			isString( results.price_range.max_price )
+				? results.price_range.max_price
+				: undefined,
```
Comment (dinhtungdu) [inline assets/js/blocks/price-filter/block.tsx:149]: I don't like these long type guards but I haven't figured out how to write recursive type predicates 😬.

**Review by albarin — APPROVED** [review]

**Pull request merged**